### DISCUSSION TITLE
how to move files/directory fron one branch to other using git extension

### DISCUSSION THREAD

**Discussion by gkunalupta** [gitextensions/gitextensions#8558] [Q&A]
I have a local repo that has 2 local branches master and slave and I want to move files from one branch to another using gitextension GUI of git. how to do that

**Comment by pmiossec** [comment]
Except if I don't understand well (and that you can't cherry pick or revert commits to do it) , that is not how git works and you won't be able to do it with Gitextensions directly. 

You could have a look at the 'worktree' feature (in the 'repository' menu) to checkout the second branch in another folder and copy/move files from one branch to the other. And do your commits accordingly.

**Comment by mstv** [comment]
- checkout the destination branch
- select the destination branch

then either:
- locate the file in the tab `File tree`
- `Reset to selected revision` using the context menu

or:
- select the file(s) in tab `Diff`
- `Reset file(s) to`|`Second: b/hash xxx` using the context menu

Cherry picking or `Open with difftool`|`Second -> Working directory` can be a means, too -- and merge or submodules, of course.
